pkg/csource/csource.go

```diff
@@ -56,7 +62,32 @@ type context struct {
 	opts      Options
 	target    *prog.Target
 	sysTarget *targets.Target
-	calls     map[string]uint64 // CallName -> NR
+	calls     map[string]uint64        // CallName -> NR
+	consts    map[string]uint64        // ConstName -> Value
+	locs      map[string]*ast.IntFlags // TypeName -> IntFlag
+}
+
+func ParseDescriptions(sysDir string, arch string) (map[string]uint64, map[string]*ast.IntFlags, error) {
+	errorBuf := new(bytes.Buffer)
+	eh := func(pos ast.Pos, msg string) {
+		fmt.Fprintf(errorBuf, "%v: %v\n", pos, msg)
+	}
+	top := ast.ParseGlob(filepath.Join(sysDir, "*.txt"), eh)
+	if top == nil {
+		return nil, nil, fmt.Errorf("failed to parse txt files:\n%s", errorBuf.Bytes())
+	}
+	consts := compiler.DeserializeConstFile(filepath.Join(sysDir, "*.const"), eh).Arch(arch)
```

Comment:
Could we use `ConstsMap` from `prog.Target` instead of parsing it here?

https://github.com/google/syzkaller/blob/4cd5bb25a2752a9a5b25597d1da34656681f07a6/prog/target.go#L65

Reply:
Ah! Fantastic :D I had no idea this even existed and didn't know I could so trivially access it from `ctx.p.Target.ConstMap`

That's a great pointer, thanks :)